libs/mock-block-dock/src/use-mock-block-props.ts

```diff
@@ -104,7 +106,11 @@ export const useMockBlockPropsNonTemporal = (
 
   const [readonly, setReadonly] = useDefaultState<boolean>(args.readonly);
 
-  const mockDatastore = useMockDatastoreNonTemporal(mockData, readonly);
+  const mockDatastore = useMockDatastoreNonTemporal(
+    mockData,
+    readonly,
+    args.simulateLatency,
+  );
```

Comment:
I've currently only simulated latency on Graph module callbacks. I'm not sure the best way to go about adding it to the Hook module or service module (or if its needed for those)

cc: @CiaranMn

Reply:
That was my reasoning too. Should I rename it to `simulateGraphLatency`? Or `simulateDatastoreLatency` or something?

I wasn't completely confident about the hook-module bit, I appreciate most of it should be the EA frontend managing things (where latency should be close to 0?) but I wasn't sure if there are still side-effects that could benefit from having an artificial slowdown somewhere between a block request and the EA

Reply:
The EA does update the entity properties in the datastore as it's editing it via the hook service, and it uses `updateEntity` [to do so](https://github.com/blockprotocol/blockprotocol/blob/db4cf8d422b881e52113aa52467d53115270e2b3/libs/mock-block-dock/src/hook-portals.tsx#L46) (just reminded myself of this) – so it should get whatever latency in that automatically.

Maybe `simulateDatastoreLatency`?

Reply:
Cool, and I've verified `updateEntity` comes from the actual `graphModuleCallbacks` as opposed to the underlying `updateEntity` implementation, so as you say it should get the latency automatically.

[Rename simulateLatency to simulateDatastoreLatency](https://github.com/blockprotocol/blockprotocol/pull/1161/commits/fc5af94920fec918061a06b0f3db7d0022f52041)